tests/test_cycle_detector.py

```
@@ -411,3 +411,126 @@ def test_dishwasher_end_spike_finishes_soon_after(mock_callbacks):
+    assert detector._end_spike_seen, "Pump-out spike should set _end_spike_seen"
+    assert detector.state == STATE_ENDING, "Should remain in ENDING after terminal spike"
+
+    # Accumulate low readings after the spike.  Fix B caps effective_off_delay at
+    # 1800 s, so the cycle must end once _time_below_threshold >= 1800 s.
+    #
+    # Caveat: the energy gate checks the recent window (last off_delay=1800 s).
+    # The pump-out spike energy (~0.4 Wh) keeps the gate closed for one extra reading
+    # after the cap fires (at exactly t_spike + 1800 s the spike is still in-window).
+    # Therefore we need 62 readings (1860 s) to push the spike out of the window and
+    # let the energy gate pass.
+    for _ in range(65):  # 65 × 30 s = 1950 s — comfortably past the gate
```

Ruff (0.15.10)

[warning] Comment contains ambiguous `×` (MULTIPLICATION SIGN). Did you mean `x` (LATIN SMALL LETTER X)?

(RUF003)